ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -37,10 +45,33 @@ public String getJobDirectory() {
 
    public void run(DatabaseClient client, InputStream csvRecords, InputStream jobControl) throws IOException {
       DataMovementManager moveMgr = client.newDataMovementManager();
+      JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
 
       try {
-         DocumentMetadataHandle documentMetadata = new DocumentMetadataHandle();
-         documentMetadata.withCollections("/jobs/"+jobId);
+          ObjectMapper objectMapper = new ObjectMapper();
+          
+          ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
+          ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
+          ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
+          
+          if(jobDef.isMissingNode()) {
+         	 throw new MarkLogicIOException("job Node is missing.");
+          }
+          String id = jobDef.path("id").asText();
+          if(id==null || id.length()==0) {
+         	 throw new MarkLogicIOException("job id cannot be empty or Null");
+          }
+  		JsonNode metadataNode = jobDef.path("metadata");
+		if (metadataNode.isMissingNode()) {
+			throw new MarkLogicIOException("metadata is  missing.");
+		}
+		String reconciliation = metadataNode.path("reconciliation").asText();
+		DocumentMetadataHandle documentMetadata = new DocumentMetadataHandle();
+		documentMetadata.withCollections("/jobs/"+id);
+         
+		DateFormat sdf = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+		Date date = new Date();
+		String dateTime = sdf.format(date);
```

Comment:
Or possibly LocalDateTime.now().toString()?  The Java API requires Java 8 as a minimum, so the Time library is available:
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html